Network config updates

rafal-ch opened the pull request:
In this PR:
- parameters for ConMan are added to the config under the `[network.conman]` section
- additional network configuration parameters are added: `bubble_timeouts`, `error_timeout` and `max_frame_size`

Commit:
Read conman configuration from config file

Commit:
Fix typo in a comment

Commit:
Add `bubble_timeouts` to network config

Commit:
Add `error_timeout` to network config

Commit:
Add `max_frame_size` to network config

Commit:
Reduce number of args for `create_rpc_builder()`

Commit:
Fix typo in the comment

Commit:
Use the config values in con man

Commit:
Make the `bubble_timeouts` enabled by default

marc-casperlabs commented on node/src/components/network/config.rs:
```diff
@@ -32,6 +32,15 @@ const DEFAULT_MAX_ADDR_PENDING_TIME: TimeDiff = TimeDiff::from_seconds(60);
+const DEFAULT_ERROR_TIMEOUT: TimeDiff = TimeDiff::from_seconds(10);
+
+/// Default max frame size
+const DEFAULT_MAX_FRAME_SIZE: u32 = 4096;
```
`max_frame_size` should be a chainspec value - any mismatch between two nodes will lead to corrupted data immediately, since `juliet` never negotiates any parameters at the start, it just assumes that both ends have pre-agreed on them.

If we do not want to make chainspec changes, we can leave this hardcoded at 4096 for now, I think.

rafal-ch replied:
Moved to chainspec in a46f0f0ea771c0b53c6341eeb146b5720195d311 and 37db66c9dd67090003f2d8cf1a81f911393e579f

marc-casperlabs commented on node/src/components/network/transport.rs:
```diff
@@ -25,31 +25,30 @@ use super::{
 };
 
-/// Creats a new RPC builder with the currently fixed Juliet configuration.
+/// Creates a new RPC builder with the currently fixed Juliet configuration.
```
This comment needs to be updated then, since the juliet configuration is no longer fixed after this PR?

rafal-ch replied:
Updated in 6a891781a543224cfcc8c5f875575bf06cd00c77

marc-casperlabs commented on resources/local/config.toml:
```diff
@@ -242,6 +242,16 @@ tarpit_chance = 0.2
 # How long peers remain blocked after they get blocklisted.
 blocklist_retain_duration = '1 minute'
 
+# Used to control if a timed-out request should make the consecutive reqeusts to fail.
```
Let's rephrase it as "Whether or not to consider a connection stuck after a single request times out, causing a termination and reconnection. It is recommended ..."

rafal-ch replied:
Updated in 6a891781a543224cfcc8c5f875575bf06cd00c77

marc-casperlabs commented on resources/local/config.toml:
```diff
@@ -242,6 +242,16 @@ tarpit_chance = 0.2
+# It is recommended to set this to `true` unless network connectivity issues are being troubleshot.
+bubble_timeouts = true
+
+# The maximum time a peer is allowed to take to receive an error.
```
I recommend changing this "a fatal error", since we always close the connection after receiving one.

rafal-ch replied:
Updated in 6a891781a543224cfcc8c5f875575bf06cd00c77

marc-casperlabs commented on resources/local/config.toml:
```diff
@@ -242,6 +242,16 @@ tarpit_chance = 0.2
+error_timeout = '10 seconds'
+
+# The maximum frame size.
+max_frame_size = 4096
```
As stated above, this needs to be a chainspec value.

rafal-ch replied:
Moved to chainspec in a46f0f0ea771c0b53c6341eeb146b5720195d311 and 37db66c9dd67090003f2d8cf1a81f911393e579f

marc-casperlabs commented on resources/local/config.toml:
```diff
@@ -252,6 +262,44 @@ blocklist_retain_duration = '1 minute'
 
 
+# ============================================
+# Configuration options for Connection Manager
```
Should either be "the connection manager" or "connection management" ?

rafal-ch replied:
Updated in 6a891781a543224cfcc8c5f875575bf06cd00c77

marc-casperlabs commented on resources/local/config.toml:
```diff
@@ -252,6 +262,44 @@ blocklist_retain_duration = '1 minute'
+# ============================================
+[network.conman]
+
+# The timeout for one TCP to be connection to be established, from a single `connect` call.
```
Let's change this to "The timeout for a single underlying TCP connection to be established.|

I'm aware this is literally copied from the comments, but those were not aimed at end users :)

rafal-ch replied:
Updated in 6a891781a543224cfcc8c5f875575bf06cd00c77

marc-casperlabs reviewed (changes requested):
Most of it is nitpicks, except three things (two of which are not mentioned in the line comments):

1. `max_frame_size` absolutely needs to be fixed or a chainspec value.
2. The existing configuration options for networking should all be re-examined, as some of them have either disappeared or repurposed and need to be renamed, e.g. `tarpit_*`.
3. All changes in configuration settings should be detailed in the `CHANGELOG`.

Commit:
Move `maximum_frame_size` from config to chainspec

Commit:
Rephrase comments

Commit:
Add conman config parameters to changelog

Commit:
Remove the stray `max_frame_size` parameter from the config

rafal-ch commented:
> Most of it is nitpicks, except three things (two of which are not mentioned in the line comments):
> 
>     1. `max_frame_size` absolutely needs to be fixed or a chainspec value.

Moved to chainspec in a46f0f0ea771c0b53c6341eeb146b5720195d311 and 37db66c9dd67090003f2d8cf1a81f911393e579f
 
>     2. The existing configuration options for networking should all be re-examined, as some of them have either disappeared or repurposed and need to be renamed, e.g. `tarpit_*`.

That's covered by a separate ticket, right? https://github.com/casper-network/casper-node/issues/4560
 
>     3. All changes in configuration settings should be detailed in the `CHANGELOG`.

Added in 4a13d723d7afdb729095fa961c28e27f297fb8e8

marc-casperlabs commented:
> That's covered by a separate ticket, right? https://github.com/casper-network/casper-node/issues/4560

No, that's for tuning the values, not removing obsolete ones. But we can merge in the current state!

marc-casperlabs commented:
https://github.com/casper-network/casper-node/pull/4607 has the removal of the network settings.
